Fill missing data_vars during concat by reindexing (#7400)
pydata/xarray: doc/whats-new.rst

  By `Benoît Bovy <https://github.com/benbovy>`_.
- Preserve original dtype on accessing MultiIndex levels (:issue:`7250`,
  :pull:`7393`). By `Ian Carroll <https://github.com/itcarroll>`_.

> - Make :py:func:`xarray.concat` more robust when concatenating variables present in some datasets but

Reviewer: ```suggestion
- :py:func:`xarray.concat` can now concatenate variables present in some datasets but
```
